### PR TITLE
Fixed onchange load issue

### PR DESCRIPTION
**Pull request by utsabpoud**
## Proposed Changes

-

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/reports/addon/components/common-actions/schedule.hbs [hunk 1]
```diff
@@ -76,7 +76,7 @@
           @options={{this.formats}}
           @selected={{this.localDeliveryRule.format.type}}
           @searchEnabled={{false}}
-          @onChange={{this.updateFormat}}
+          @onChange={{fn (mut this.localDeliveryRule.format.type)}}
```
[the remark this answers is not in the document]
Comment (utsabpoud) [reply packages/reports/addon/components/common-actions/schedule.hbs:79]: formatOption test cases cover this case as well 
```format options - fallback to `exportFileTypes```

[the remark this answers is not in the document]
Comment (jkusa) [reply packages/reports/addon/components/common-actions/schedule.hbs:79]: Can we just update the action in the TS file? 

```
this.localDeliveryRule.format.type = type;
```

Also, can we please add a test?

[the remark this answers is not in the document]
Comment (utsabpoud) [reply packages/reports/addon/components/common-actions/schedule.hbs]: Added Unit test case for the format selection and removed type in delivery rule in ts file.

**Comment by daniellevinson** [comment]
@utsabpoud please rebase I've merged changes to the schedule component

**Review by cythrawll — APPROVED** [review]

File: packages/reports/tests/acceptance/schedule-modal-test.js [hunk 2]
```diff
@@ -200,9 +200,11 @@ module('Acceptance | Navi Report Schedule Modal', function (hooks) {
...
     );
 
-    assert
-      .dom('.schedule__modal-format-trigger .ember-power-select-selected-item')
-      .hasText('csv', 'Format field is populated by existing delivery rule');
+    assert.equal(
+      $('.schedule__modal-format-trigger .ember-power-select-selected-item').text().trim(),
+      { type: 'csv' },
+      'Format field is populated by existing delivery rule'
+    );
```
Comment (daniellevinson) [inline packages/reports/tests/acceptance/schedule-modal-test.js:207]: aren't we converting from `{ type: 'csv' }` to just `'csv'`? shouldn't the old assert still work here?

File: packages/reports/tests/integration/components/common-actions/schedule-test.js [hunk 3]
```diff
@@ -5,11 +5,13 @@ import { setupRenderingTest } from 'ember-qunit';
...
 import { nativeMouseUp } from 'ember-power-select/test-support/helpers';
+import { selectChoose } from 'ember-power-select/test-support';
+//@ts-ignore
+import { animationsSettled } from 'ember-animated/test-support';
 import config from 'ember-get-config';
-
 const DeliveryRule = {
   frequency: 'Week',
-  format: { type: 'csv' },
+  format: 'csv',
```
Comment (daniellevinson) [inline packages/reports/tests/integration/components/common-actions/schedule-test.js:14]: maybe I'm not in the loop but will our web services support this change?

File: packages/reports/tests/integration/components/common-actions/schedule-test.js [hunk 4]
```diff
@@ -351,6 +353,29 @@ module('Integration | Component | common actions/schedule', function (hooks) {
...
+
+    await selectChoose('.schedule__modal-format-trigger', 'csv');
+
+    await animationsSettled();
+
+    assert.equal(
+      $('.schedule__modal-format-trigger .ember-power-select-selected-item').text().trim(),
+      'csv',
+      'Schedule format  CSV shouldbe selected on change '
+    );
```
Comment (daniellevinson) [inline packages/reports/tests/integration/components/common-actions/schedule-test.js:376]:
```suggestion
    assert
      .dom('.schedule__modal-format-trigger .ember-power-select-selected-item')
      .hasText('csv', 'CSV format should be selected on change');
```

**Comment by daniellevinson** [comment]
can we add a test that we're not getting duplicate csv option?

File: packages/reports/addon/components/common-actions/schedule.ts [hunk 5]
```diff
@@ -136,7 +136,7 @@ export default class ScheduleActionComponent extends Component<Args> {
    */
   _createNewDeliveryRule() {
     return this.store.createRecord('delivery-rule', {
-      format: { type: this.formats.firstObject },
+      format: this.formats.firstObject,
```
Comment (daniellevinson) [inline packages/reports/addon/components/common-actions/schedule.ts:139]: should we have a return type for this method?

File: packages/reports/tests/integration/components/common-actions/schedule-test.js (hunk 4 above)
Comment (utsabpoud) [reply packages/reports/tests/integration/components/common-actions/schedule-test.js:376]: Resolved

**Review by utsabpoud — COMMENTED** [review]
@daniellevinson Since I don't have write access, can u merge this PR?.

File: packages/reports/addon/components/common-actions/schedule.ts (hunk 5 above)
Comment (kevinhinterlong) [reply packages/reports/addon/components/common-actions/schedule.ts:139]: Looks like `DeliveryRule` has some incorrect TS saying this is a string ([WS DeliveryRule Model](https://github.com/yavin-dev/framework/blob/71358ed71cbcf012bb6d97e744abc2076b7b44a6/packages/webservice/models/src/main/kotlin/com/yahoo/navi/ws/models/beans/DeliveryRule.kt#L84) says `{ format: { type: 'csv' }}` is correct.) We should make a `delivery-format` fragment like we do for [scheduling-rules fragment](https://github.com/yavin-dev/framework/blob/master/packages/core/addon/models/fragments/scheduling-rules.ts) and use it like this in the [delivery-rule model](https://github.com/yavin-dev/framework/blob/71358ed71cbcf012bb6d97e744abc2076b7b44a6/packages/core/addon/models/delivery-rule.ts#L47-L48)

**Review by kevinhinterlong — CHANGES_REQUESTED** [review]

File: packages/core/addon/models/fragments/delivery-format.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+/**
+ * Copyright 2021, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ */
+import { attr } from '@ember-data/model';
+import Fragment from 'ember-data-model-fragments/fragment';
+
+export default class DeliveryFormatFragment extends Fragment {
+  @attr('string')
+  type!: string;
```
Comment (jkusa) [inline packages/core/addon/models/fragments/delivery-format.ts:10]:
```suggestion
  declare type: string;
```

File: packages/core/addon/models/delivery-rule.ts [hunk 7]
```diff
@@ -47,8 +48,8 @@ export default class DeliveryRuleModel extends Model.extend(Validations) {
   @fragment('fragments/scheduling-rules', { defaultValue: () => ({ mustHaveData: false }) })
   schedulingRules!: SchedulingRuleFragment;
 
-  @attr()
-  format!: string;
+  @fragment('fragments/delivery-format')
+  deliveryFormat!: DeliveryFormatFragment;
```
Comment (jkusa) [inline packages/core/addon/models/delivery-rule.ts:52]: The backend field name is `format`

https://github.com/yavin-dev/framework/blob/4cd0d8f5353dd8c2d0c05f440e9add32719dc8b7/packages/webservice/models/src/main/kotlin/com/yahoo/navi/ws/models/beans/DeliveryRule.kt#L84

**Review by jkusa — CHANGES_REQUESTED** [review]

File: packages/reports/tests/integration/components/common-actions/schedule-test.js [hunk 8]
```diff
@@ -351,6 +353,24 @@ module('Integration | Component | common actions/schedule', function (hooks) {
...
+    this.set('model', TestModel);
+    await render(TEMPLATE);
+
+    await click('.schedule-action__button');
+
+    await click('.schedule__modal-format-trigger');
+
+    await selectChoose('.schedule__modal-format-trigger', 'csv');
+
+    await animationsSettled();
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/common-actions/schedule-test.js:370]: Is this needed?

**Review by kevinhinterlong — APPROVED** [review]

**Comment by daniellevinson** [comment]
can we also fix the duplicate csv option for reports and add a test for that case?
Screen Shot 2021-09-08 at 12 02 02 PM

File: packages/reports/tests/integration/components/common-actions/schedule-test.js [hunk 9]
```diff
@@ -351,6 +350,21 @@ module('Integration | Component | common actions/schedule', function (hooks) {
...
+    this.set('model', TestModel);
+    await render(TEMPLATE);
+
+    await click('.schedule-action__button');
+
+    await click('.schedule__modal-format-trigger');
+
+    assert.dom('.schedule__modal-format-trigger').hasText('csv', 'Schedule format should have correct default option');
+    config.navi.FEATURES.exportFileTypes = originalFlag;
+  });
```
Comment (jkusa) [inline packages/reports/tests/integration/components/common-actions/schedule-test.js:367]:
```suggestion
  });
  
```

**Comment by daniellevinson** [comment]
it happens when `exportFileTypes: ['csv', 'pdf', 'png', 'gsheet']`

**Comment by utsabpoud** [comment]
> it happens when `exportFileTypes` includes `'csv'`

when `exportFileTypes` is present, we need to set the default selection `'csv'` format right? since default value is concatenated in format array while loading the  `PowerSelect`

**Comment by utsabpoud** [comment]
> > it happens when `exportFileTypes` includes `'csv'`
> 
> when `exportFileTypes` is present, we need to set the default selection `'csv'` format right? since default value is concatenated in format array while loading the `PowerSelect`. Refernce [link](https://github.com/yavin-dev/framework/blob/master/packages/reports/addon/components/common-actions/schedule.ts#L104)

**Review by jkusa — APPROVED** [review]
